### PR TITLE
11.0 mig account invoice fiscal position update

### PR DESCRIPTION
**Pull request by ghost**
Migrate to 11.0

CLA Bot will probably give a warning, because my github acocunt is not known, mailed cla@ about this.
@RoelAdriaans-B-informed is me with signed cla, but company account.

### CODE REVIEW

File: account_invoice_fiscal_position_update/__manifest__.py [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 # © 2011-2014 Julius Network Solutions SARL <contact@julius.fr>
 # © 2014-2016 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+# © 2018 Roel Adriaans <roel@road-support.nl>
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
 
 {
     'name': 'Invoice Fiscal Position Update',
-    'version': '10.0.1.0.1',
+    'version': '11.0.1.0.1',
```
Comment (pedrobaeza) [inline account_invoice_fiscal_position_update/__manifest__.py]: 11.0.1.0.0

File: account_invoice_fiscal_position_update/README.rst [hunk 2]
```diff
@@ -38,10 +41,13 @@ Credits
 Contributors
 ------------
 
+* Roel Adriaans <roel@road-support.nl>
```
Comment (pedrobaeza) [inline account_invoice_fiscal_position_update/README.rst]: Please put you in last position, as this history is supposed to be from beginning to recent contributions

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by RoelAdriaans** [comment]
@pedrobaeza Changes Done.

**Review by pedrobaeza — APPROVED** [review]
Please squash your 3 migration commits in one.

File: account_invoice_fiscal_position_update/models/account_invoice.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        res = {}
+        lines_without_product = []
+        fp = self.fiscal_position_id
+        inv_type = self.type
+        for line in self.invoice_line_ids:
+            if line.product_id:
+                product = line.product_id
+                if inv_type in ('out_invoice', 'out_refund'):
+                    account = (
+                        product.property_account_income_id or
```
Comment (etobella) [inline account_invoice_fiscal_position_update/models/account_invoice.py]: You should ensure the company of the property for multicompany environments 
`with_context(force_company=self.company_id.id)`

File: account_invoice_fiscal_position_update/models/account_invoice.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                        product.categ_id.property_account_income_categ_id)
+                    # M2M fields don't have an option 'company_dependent=True'
+                    # so we need per-company post-filtering
+                    taxes = product.taxes_id.filtered(
+                        lambda tax: tax.company_id == self.company_id)
+                else:
+                    account = (
+                        product.property_account_expense_id or
```
Comment (etobella) [inline account_invoice_fiscal_position_update/models/account_invoice.py:44]: The same

**Review by etobella — CHANGES_REQUESTED** [review]
Just some comments in order to make it usable in multicompany environments.

**Comment by JordiBForgeFlow** [comment]
@RoelAdriaans-old are you planning to consider the latest changes? Or should we better make a pull request that supersedes this one?

**Comment by RoelAdriaans** [comment]
@jbeficent Sorry forgot about it.. 
I'll apply the changes suggested by @etobella  , can you validate that this is correct?

**Review by etobella — APPROVED** [review]
LGTM :+1:

**Comment by HviorForgeFlow** [comment]
I noticed test_inv_fiscal_pos_update is failing in my CI. I think, it is because I run test after install l10n location.

```
2018-08-20 16:18:27,215 48 INFO test odoo.modules.module: odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update running tests.
2018-08-20 16:18:27,215 48 INFO test odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update: test_fiscal_position_id_change (odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update.TestProductIdChange)
2018-08-20 16:18:27,479 48 ERROR test odoo.sql_db: bad query: b'INSERT INTO "account_account" ("id", "deprecated", "user_type_id", "company_id", "reconcile", "code", "name", "centralized", "create_uid", "write_uid", "create_date", "write_date") VALUES(nextval(\'account_account_id_seq\'), false, 14, 1, true, \'710000\', \'customer export account\', false, 1, 1, (now() at time zone \'UTC\'), (now() at time zone \'UTC\')) RETURNING id'
ERROR: duplicate key value violates unique constraint "account_account_code_company_uniq"
DETAIL:  Key (code, company_id)=(710000, 1) already exists.

2018-08-20 16:18:27,487 48 ERROR test odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update: ERROR
2018-08-20 16:18:27,487 48 INFO test odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update: ======================================================================
2018-08-20 16:18:27,487 48 ERROR test odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update: ERROR: test_fiscal_position_id_change (odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update.TestProductIdChange)
2018-08-20 16:18:27,487 48 ERROR test odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update: Traceback (most recent call last):
2018-08-20 16:18:27,487 48 ERROR test odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update: `   File "/opt/odoo/auto/addons/account_invoice_fiscal_position_update/tests/test_inv_fiscal_pos_update.py", line 47, in test_fiscal_position_id_change
2018-08-20 16:18:27,488 48 ERROR test odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update: `     'reconcile': True,
2018-08-20 16:18:27,488 48 ERROR test odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update: `   File "/opt/odoo/custom/src/odoo/odoo/models.py", line 3373, in create
2018-08-20 16:18:27,488 48 ERROR test odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update: `     record = self.browse(self._create(old_vals))
2018-08-20 16:18:27,488 48 ERROR test odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update: `   File "/opt/odoo/custom/src/odoo/odoo/models.py", line 3466, in _create
2018-08-20 16:18:27,488 48 ERROR test odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update: `     cr.execute(query, tuple(u[2] for u in updates if len(u) > 2))
2018-08-20 16:18:27,488 48 ERROR test odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update: `   File "/opt/odoo/custom/src/odoo/odoo/sql_db.py", line 155, in wrapper
2018-08-20 16:18:27,488 48 ERROR test odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update: `     return f(self, *args, **kwargs)
2018-08-20 16:18:27,488 48 ERROR test odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update: `   File "/opt/odoo/custom/src/odoo/odoo/sql_db.py", line 232, in execute
2018-08-20 16:18:27,488 48 ERROR test odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update: `     res = self._obj.execute(query, params)
2018-08-20 16:18:27,488 48 ERROR test odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update: ` psycopg2.IntegrityError: duplicate key value violates unique constraint "account_account_code_company_uniq"
2018-08-20 16:18:27,488 48 ERROR test odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update: ` DETAIL:  Key (code, company_id)=(710000, 1) already exists.
2018-08-20 16:18:27,488 48 ERROR test odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update: ` 
2018-08-20 16:18:27,489 48 INFO test odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update: Ran 1 test in 0.272s
2018-08-20 16:18:27,489 48 ERROR test odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update: FAILED
2018-08-20 16:18:27,489 48 INFO test odoo.addons.account_invoice_fiscal_position_update.tests.test_inv_fiscal_pos_update:  (errors=1)
2018-08-20 16:18:27,489 48 ERROR test odoo.modules.module: Module account_invoice_fiscal_position_update: 0 failures, 1 errors
```

**Comment by pedrobaeza** [comment]
Tests should search for an existing account first instead of creating it, or use it a weird code that is not going to choke (example: `account_invoice_fiscal_position_update_710000`)

File: account_invoice_fiscal_position_update/tests/test_inv_fiscal_pos_update.py [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+                self.env.ref('account.data_account_type_receivable').id)],
+            limit=1)
+        self.account_revenue = self.env['account.account'].search(
+            [('user_type_id', '=',
+                self.account_user_type.id)],
+            limit=1)
+
+    def test_fiscal_position_id_change(self):
+        partner = self.res_partner_model.create(dict(name="George"))
+        account_export_id = self.account_model.sudo().create({
```
Comment (HviorForgeFlow) [inline account_invoice_fiscal_position_update/tests/test_inv_fiscal_pos_update.py:43]: account_export_id = self.account_model.sudo().search(
            [('code', '=', '710000')], limit=1
        )
        if not account_export_id:
            account_export_id = self.account_model.sudo().create({
                'code': "710000",
                'name': "customer export account",
                'user_type_id': self.account_user_type.id,
                'reconcile': True,
            })

**Review by HviorForgeFlow — CHANGES_REQUESTED** [review]

File: account_invoice_fiscal_position_update/tests/test_inv_fiscal_pos_update.py (hunk 5 above)
Comment (RoelAdriaans) [reply account_invoice_fiscal_position_update/tests/test_inv_fiscal_pos_update.py:43]: :+1: I'll try to make the changes this weekend

**Review by CasVissers-360ERP — APPROVED** [review]
Functional test done. LGTM

**Comment by pedrobaeza** [comment]
@RoelAdriaans can you please finish this?

**Comment by pedrobaeza** [comment]
I have fixed the test creating an unique account code, so I merge.

**Pull request merged**